lib/avromatic/model/attribute/abstract_timestamp.rb

```diff
@@ -0,0 +1,26 @@
+module Avromatic
+  module Model
+    module Attribute
+
+      # This subclass of Virtus::Attribute is used to truncate timestamp values
+      # to the supported precision.
+      class AbstractTimestamp < Virtus::Attribute
+        def coerce(value)
+          return value if value.nil? || value_coerced?(value)
+
+          value.is_a?(Time) ? coerce_time(value) : value
+        end
+
+        def value_coerced?(_value)
+          raise 'subclass must implement `value_coerced?`'
```

Comment:
We would get some more useful context if the message was `"#{__method__} must be overridden by #{self.class.name}"`